deployer/orb.yaml

```
@@ -1,6 +1,85 @@
+              tag = '<<parameters.verify_version>>'
+            v = Version(tag)
+            assert tag == str(v), f'Original tag ({tag}) should have been {str(v)}'
+            assert v.is_devrelease or not v.local, f'tag should not contain \'+{v.local}\' unless dev version'
+
+  verify-tag-matches-deploy:
+    parameters:
+      folder:
+        default: '.'
+        description: >
+          Path to the folder containing the Python package.
+        type: string
+      verify_version:
+        default: CIRCLE_TAG
+        description: >
+          If set to a non-empty value, fails the deployment if the version
+          being deployed does not match. Can be used as a last-step sanity
+          check.
+        type: string
+    steps:
+      - run:
+          name: verify tag builds match deploy version
+          command: |
+            if [[ ! -z "${<<parameters.verify_version>>}" ]]; then
+                code_version=$(find "<<parameters.folder>>/dist" -iname '*.tar.gz' | awk -F- '{sub(/\.tar\.gz/, ""); print $NF}')
+                code_version_alt="${code_version/.dev/-dev.}"
+                if [[ <<parameters.verify_version>> != ${code_version} &&
+                      <<parameters.verify_version>> != ${code_version_alt} ]]; then
+                    echo "Version ${code_version} does not match tag ${<<parameters.verify_version>>}"
+                    exit 1
+                fi
+            fi
+
+  upload:
+    parameters:
+      folder:
+        default: '.'
+        description: >
+          Path to the folder containing the Python package.
+        type: string
+      gemfury_token:
+        default: GEMFURY_TOKEN
+        description: >
+          Name of the env var containing the Gemfury token.
+        type: env_var_name
+    steps:
+      - run: ls "<<parameters.folder>>/dist" | xargs -I{} curl -F package=@<<parameters.folder>>/dist/{} https://${<<parameters.gemfury_token>>}@push.fury.io/talkiq/
```

This reminds me that we should actually make a ticket for addressing silent failures when we fail to upload to gemfury. For instance, we might silently fail with a message like this:
```
#!/bin/bash -eo pipefail
ls "core/dist" | xargs -I{} curl -F package=@core/dist/{} https://${GEMFURY_TOKEN}@push.fury.io/talkiq/
~~> Processing package upload
    voiceai-formatter-core-6.1.0.dev10.tar.gz ... ok
~~> Processing package upload
    voiceai_formatter_core-6.1.0.dev10-py2.py3-none-any.whl ... failed
CircleCI received exit code 0
```

---

https://switchcomm.atlassian.net/browse/VAE-2276